cardano-ledger-test/src/Test/Cardano/Ledger/Properties.hs

```diff
@@ -0,0 +1,794 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE TupleSections #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Test.Cardano.Ledger.Properties where
```

Comment:
Are the generators in this module intended to be used elsewhere? If so, I'd prefer we split apart the generators (test infrastructure) from the properties themselves (which should not be included in the library)

Reply:
Generators in this module only work in `RWST GenEnv () GenState Gen` monad, so as it stands right now they can't be used anywhere else.

> This is now adding another new method atop the existing one and the one we've asked the Obsidian folks to build.

I believe this is orthogonal to what Obsidian folks are doing. This is intended for testing a particular component, rather than creating a framework that will test all transition rules from epoch to epoch. What do you mean by this: "atop the existing one"? I don't really see what is the bottom method it is building atop off?

> I'd prefer we split apart the generators (test infrastructure) from the properties themselves (which should not be included in the library)

I certainly agree. This module was originally named `Properties`, but I think it should be renamed to something like `Generators.Ledger` or something along those lines. I guess the problem I see is there no concrete structure or documented plan to what `cardano-ledger-test` should be. Has it been outlined anywhere?